src/lib/classes/NetworkManager.ts

```diff
@@ -0,0 +1,104 @@
+import { writable, type Writable } from "svelte/store";
+import type { Connection, NetworkState, Node, NodeType } from "$lib/interfaces/network";
+
+export class NetworkStore implements Writable<NetworkState> {
+	private store = writable<NetworkState>({ nodes: [], connections: [] });
+
+	set: Writable<NetworkState>["set"] = this.store.set;
+	update: Writable<NetworkState>["update"] = this.store.update;
+	subscribe: Writable<NetworkState>["subscribe"] = this.store.subscribe;
+
+	private _nodes: Node[] = [];
+	private _connections: Connection[] = [];
+	private nodeMap = new Map<number, Node>();
+	private counter = 0;
+
+	get nodes() {
+		return this._nodes;
+	}
+
+	get connections() {
+		return this._connections;
+	}
+
+	get networkState(): NetworkState {
+		return {
+			nodes: this.nodes,
+			connections: this.connections,
+		};
+	}
+
+	deleteConnection(connection: Connection) {
+		this._connections = this._connections.filter(
+			(c) => c.source.id !== connection.source.id || c.target.id !== connection.target.id,
+		);
+
+		this.fastUpdate();
+	}
+
+	deleteNode(nodeId: number) {
+		const node = this.getNode(nodeId);
+
+		if (node) {
+			this._nodes = this._nodes.filter((n) => n.id !== nodeId);
+			this.nodeMap.delete(nodeId);
+		}
+
+		this.fastUpdate();
+	}
+
+	addConnection(input: { source: Node; target: Node }) {
+		// Ensure that the two nodes are not the same
```

Comment:
Delete this

Reply:
yeah i agree the code is self documenting

Reply:
![image](https://tenor.com/view/pepsi-bottle-coca-cola-glas-i-dont-give-a-damn-pepsi-coca-cola-gif-14899661)